thunx-predicates-solr/src/main/java/eu/xenit/contentcloud/thunx/predicates/solr/SolrUtils.java

```diff
@@ -0,0 +1,14 @@
+package eu.xenit.contentcloud.thunx.predicates.solr;
+
+import eu.xenit.contentcloud.thunx.predicates.model.ThunkExpression;
+
+public class SolrUtils {
+
+    public static String from(ThunkExpression<Boolean> thunk/*, PathBuilder<?> entityPath*/) {
```

Comment:
Does the `SolrConverter` not need to know for which entity we're doing a query ? The thunks/expressions use `entity` as a placeholder, but is only applied on a specific entity.

In the querydsl implementation, this information is captured in this `PathBuilder<?>`.

Reply:
No, I don't think so.  From the looks of the querydsl converter you need to know the subject in order to build a "correct" predicate expression.

In solr, additional sync'ed attributes are under the control of the application actually and its there where the attributes are defined (by defining an [AttributeProvider](https://github.com/paulcwarren/abac_spike/blob/8304a08e9d417485aca2160d62f11101e864fdcd/abac-spike-2/src/main/java/com/example/demo/ABACSpike2Application.java#L54) bean).  So, actually, I am not even sure this solr converter should be part of the set of thunx libraries.  It should probably just be code in the application, or an application library.